lib/utils/ast-utils.js

```diff
@@ -35,6 +35,79 @@ exports.init = function (sourceCode) {
  */
 exports.isASTNode = nodeSchemaValidator.compile (astNodeSchema);
 
+/**
+ * @param {Object} node The node to check
+ * @param {String} type The type of the node
+ * @returns {Boolean} true if the given node is the right type
+ */
+function isNodeType (node, name) {
```

Comment:
Please add a validation at the start: `throwIfInvalidNode(node, '<function name>');`. Method is already defined, just need to use it.
See [example](https://github.com/duaraghav8/Solium/blob/master/lib/utils/ast-utils.js#L44).

Reply:
since all your util methods call `isNodeType`, only this function needs to call this validation method.